kubectl-plugin/src/resources/pool.rs

```diff
@@ -1,26 +1,47 @@
 use crate::{
     operations::{Get, List},
-    resources::PoolId,
+    resources::{utils, OutputFormat, PoolId},
     rest_wrapper::RestClient,
 };
-use anyhow::Result;
 use async_trait::async_trait;
+use prettytable::Row;
 use structopt::StructOpt;
-
 /// Pools resource.
 #[derive(StructOpt, Debug)]
-pub struct Pools {}
+pub struct Pools {
+    output: OutputFormat,
+}
 
 #[async_trait(?Send)]
 impl List for Pools {
-    async fn list() -> Result<()> {
-        let pools = RestClient::client()
-            .pools_api()
-            .get_pools()
-            .await
-            .map_err(|e| anyhow::anyhow!("Failed to get pools. Error {}", e))?;
-        println!("{:?}", pools);
-        Ok(())
+    type Format = OutputFormat;
+    async fn list(output: &Self::Format) {
+        match RestClient::client().pools_api().get_pools().await {
+            Ok(pools) => match output.to_lowercase().trim() {
+                "" => {
+                    // Show the tabular form if output format is not specified.
+                    let rows: Vec<Row> = utils::create_pool_rows(pools);
+                    utils::table_printer((&*utils::POOLS_HEADERS).clone(), rows);
+                }
+                utils::YAML_FORMAT => {
+                    // Show the YAML form output if output format is YAML.
+                    let s = serde_yaml::to_string(&pools).unwrap();
+                    println!("{}", s);
+                }
+                utils::JSON_FORMAT => {
+                    // Show the JSON form output if output format is JSON.
+                    let s = serde_json::to_string(&pools).unwrap();
+                    println!("{}", s);
+                }
+                _ => {
+                    // Incase of invalid output format, show error and gracefully end.
+                    println!("Output not supported for {} format", output);
+                }
+            },
+            Err(e) => {
+                println!("Failed to list pools. Error {}", e)
+            }
```

Comment:
Suggestion: This pattern is repeated for both volumes and pools. A generic function could be implemented for this instead. I guess in this case you would require a "create_rows" trait that would be implemented by both volumes and pools.

Reply:
Added a few more changes viz, GetHeaderRow, cleaned some code, added comments on top of your changes. Thank you.